signin_parser_spec.rb

```diff
@@ -8,24 +8,35 @@
     @login = 'teacher_128'
     @password = 'teacher_128_pass'
     @enter = "Вхід"
-    @radio = 'depart_subj_spring'
     @quit = "Вийти"
   end
 
-  it "should be signed in, parsed and signed_out" do
+  it "should be signed in and sign out" do
     signed = @signin_parse.sign_in(@url, @login, @password, @enter)
-    expect(signed.class).to eq Capybara::Node::Element
-    parsed_items = @signin_parse.parse("TeacherSubject", @radio)
-    expect(parsed_items.size).to eq 69
-    parsed_items.each do |item|
-      expect(item).to match /^Дисципліна_.+/
-    end
+    expect(signed.class).to eq Capybara::Session
+    expect(signed.find('a', text: @quit).class).to eq Capybara::Node::Element
     @signin_parse.sign_out(@quit)
+    expect(signed.find_button(@enter).class).to eq Capybara::Node::Element
   end
 
   it "should be valid login and password" do
     signed = @signin_parse.sign_in(@url, '', '', @enter)
-    expect(signed.class).to eq Capybara::Node::Element
+    begin
+      quit_found = signed.find('a', text: @quit)
+    rescue Capybara::ElementNotFound
+      quit_found = false
+    end
```

Comment:
That's pretty ugly. You should check Logout button in your login method, if you find one after submit login, it means your creds are good, if not you should raise some specific error and that's what you should be checking here, that specific error raised with wrong creds.